### PR TITLE
fix(ui-shell): add menubar role to platform header

### LINKED ISSUE #1270: Add menubar role for Platform Header dropdown
Looking at the WAI ARIA practices site, there is a reference to a menu or menu bar that we will need to add to our platform header: https://www.w3.org/TR/wai-aria-practices/#menu. There is also an example implementation available here: https://www.w3.org/TR/wai-aria-practices/examples/menubar/menubar-1/menubar-1.html

This issue is for updating the existing markup with the correct a11y-related roles or attributes, adjusting any styles as needed.

### PR DESCRIPTION
**Pull request by joshblack**
Closes #1270

Add support for `menubar` role in `platform-header`

#### Changelog

**New**

**Changed**

- Update `platform-header` markup and styles for adding in this a11y role

**Removed**

### CODE REVIEW

**Review by asudoh — COMMENTED** [review]
The markup part LGTM 👍. Great to have reviews from others for the Sass part, and to help that, great to have articulation on what caused the style structure change. Thanks @joshblack!

File: src/components/ui-shell/_platform-header.scss [hunk 1]
```diff
@@ -82,100 +85,91 @@
...
   }
 
-  .#{$prefix}--dropdown-item__menu-item {
-    width: units(25);
-    height: units(6);
+  .#{$prefix}--platform-header__menu-title[role='menuitem'][aria-expanded='true'] {
+    background-color: $ibm-colors__gray--80;
+    // Note: needs to be higher than menu. Adding 1 here instead of moving to
+    // the next level.
+    z-index: #{z('header') + 1};
```
Comment (alisonjoseph) [inline src/components/ui-shell/_platform-header.scss:138]: Didn't know we could do this with z-index! nice!

**Review by alisonjoseph — APPROVED** [review]
Looks good to me! Thanks for adding in all the aria stuff 😄

**Comment by joshblack** [comment]
@asudoh most of the changes are in order to support the menubar aria role, in addition to tying our selectors into aria roles or attributes just so that if markup does not have those attributes then the styles won't work 👍

**Review by aledavila — APPROVED** [review]

**Pull request merged**